metadata-ingestion/src/datahub/ingestion/source/dbt_manifest.py

```diff
@@ -48,50 +57,217 @@ class DBTManifestConfig(ConfigModel):
+            dbtNode.upstream_urns = []
+
+        if dbtNode.materialization != 'ephemeral':
+            dbtNode.columns = get_columns(catalog[dbtNode.dbt_name])
+        else:
+            dbtNode.columns = []
+
+        dbtNode.datahub_urn = get_urn_from_dbtNode(dbtNode.database, dbtNode.schema, dbtNode.relation_name, platform, environment)
+        
+        dbt_entities.append(dbtNode)
+
+    return dbt_entities
+
+
+def loadManifestAndCatalog(manifest_path, catalog_path, platform, environment) -> list[DBTNode]:
+    manifest_nodes = list[DBTNode]
+    
+    with open(manifest_path, "r") as manifest:
+        with open(catalog_path, "r") as catalog: 
+            dbt_manifest_json = json.load(manifest)
+            dbt_catalog_json = json.load(catalog)
+
+            manifest_nodes = dbt_manifest_json['nodes']
+            manifest_sources = dbt_manifest_json['sources']
+
+            all_manifest_entities = manifest_nodes | manifest_sources
+
+            catalog_nodes = dbt_catalog_json['nodes']
+            catalog_sources = dbt_catalog_json['sources']
+
+            all_catalog_entities = catalog_nodes | catalog_sources
+
+            logger.info('loading {} nodes, {} sources'.format(len(all_manifest_entities), len(all_catalog_entities)))
+    
+            nodes = extract_dbt_entities(all_manifest_entities, all_catalog_entities, platform, environment)
+
+            return nodes
+
+def get_urn_from_dbtNode(database: str, schema: str, relation_name: str, platform: str, env:str) -> str:
+    db_fqn = f"{database}.{schema}.{relation_name}".replace('"', '')
+    return f"urn:li:dataset:(urn:li:dataPlatform:{platform},{db_fqn},{env})"
+
+def get_custom_properties(node: DBTNode) -> List[dict[str,str]]:
+    tags = {}
+    tags['dbt_node_type'] = node.node_type
+    tags['materialization'] = node.materialization
+    tags['dbt_file_path'] = node.dbt_file_path
+    return tags
+
+def get_upstreams(upstreams: List[str], all_nodes, platform: str, environment: str) -> List[str]:
+    upstream_urns = []
+
+    for upstream in upstreams:
+        upstream_node = all_nodes[upstream]
+
+        upstream_urns.append(get_urn_from_dbtNode(
+            upstream_node['database'],
+            upstream_node['schema'],
+            upstream_node['relation_name'],
+            platform,
+            environment
+        ))
+
+    return upstream_urns
+
+def get_upstream_lineage(upstream_urns: List[str]) -> UpstreamLineage:
+    ucl: List[uc] = []
+
+    for dep in upstream_urns:
+        uc = UpstreamClass(
+        dataset=dep, 
+        auditStamp=AuditStamp(
+            actor="urn:li:corpUser:dbt_executor",
+            time=1581407189000 # replace with system timestamp etc.
+        ),
+        type="TRANSFORMED"
+        )
+        ucl.append(uc)
+
+    ulc = UpstreamLineageClass(
+        upstreams = ucl
+    )
+
+    return ulc
+
+_field_type_mapping = {
+    'boolean': BooleanTypeClass,
+    'date': StringTypeClass, # Is there no DateTypeClass?
```

Comment:
There should be a DateTypeClass - I added it last week in this PR https://github.com/linkedin/datahub/pull/2201

You might need to rerun `(cd .. && ./gradlew :metadata-events:mxe-schemas:build)` and `./scripts/codegen.sh` in order to see it?

Reply:
More broadly, is there a list of all type names that DBT might produce? - would be good to get full coverage right off the bat

Reply:
This is actually a challenge, https://docs.getdbt.com/reference/artifacts/catalog-json 

DBT doesn't publish their data types as a list, I'm betting that the datatypes are probably just scraped from the database.  In fact the one thing that DBT does poorly (or just doesn't do at all) is take any active interest in data types, all data types are determined by querying the underlying data plane (postgress in this case and Redshift in the other case I deal with).

I'll dig into this a bit more, but it may not be something with an easy answer in datahub.

Reply:
If you're interested @hsheth2 this is where we get the dbt sample output files.  https://docs.getdbt.com/reference/artifacts/catalog-json

Reply:
One final note, I'm digging through the DBT generate code trying to figure out how those fields are populated... As I mentioned above I'm expecting to find it's a straight text representation from the storage layer (IE Redshift / Postgres).  If that's the case we'll have to come up with something more durable than a straight 1-1 mapping

Reply:
In that case, the mapping doesn't need to be perfect - you're also populating the nativeDataType field, which should serve as a decent fallback in case the mapping doesn't have the actual typ

Reply:
> In that case, the mapping doesn't need to be perfect - you're also populating the nativeDataType field, which should serve as a decent fallback in case the mapping doesn't have the actual type

In that case I'll just tweak the types a bit (peel off parantheses etc) and otherwise leave as is. 

maybe a future piece of work would be to try and use SQLAlchemy type mappings as presumably they've already got this worked out.

Reply:
Sounds good

SQLAlchemy does have these mappings, but they're buried inside the internals and vary between database drivers, so it likely won't be straightforward - agree that we should punt on that as future work